ui-participant/src/Navbar.tsx

```diff
@@ -31,6 +31,13 @@ export default function Navbar(props: NavbarProps) {
     localContent
   } = usePortalEnv()
 
+  useEffect(() => {
+    if (profile && profile.preferredLanguage && selectedLanguage !== profile.preferredLanguage) {
+      changeLanguage(profile.preferredLanguage)
+      reloadPortal()
+    }
+  }, [profile?.preferredLanguage, selectedLanguage, changeLanguage])
+
```

Comment:
So, if you logged into a user who has never been to Juniper before, they won't have the right preferred language in session storage. It takes a full refresh for it to populate - for some reason. Everything related to managing language state seems to be a can of worms, so this was my perhaps hamfisted solution to make sure it is synced on initial load.